tensorflow/python/keras/layers/wrappers.py

```diff
@@ -208,27 +227,42 @@ def step(x, _):
           step,
           inputs,
           initial_states=[],
-          input_length=row_lengths[0] if is_ragged_input else input_shape[1],
+          input_length=row_lengths[0] if is_ragged_input else input_length,
           mask=mask,
           unroll=False)
-      y = K.maybe_convert_to_ragged(is_ragged_input, outputs, row_lengths)
+      y = nest.map_structure(
+          lambda output: K.maybe_convert_to_ragged(is_ragged_input, output, row_lengths),
+          outputs)
     else:
       # No batch size specified, therefore the layer will be able
       # to process batches of any size.
       # We can go with reshape-based implementation for performance.
-      if isinstance(inputs, ragged_tensor.RaggedTensor):
-        y = self.layer(inputs.values, **kwargs)
-        y = ragged_tensor.RaggedTensor.from_row_lengths(
-            y,
-            inputs.nested_row_lengths()[0])
+      is_ragged_input = nest.map_structure(lambda x: isinstance(x, ragged_tensor.RaggedTensor),
+                                           inputs)
+      is_ragged_input = generic_utils.to_list(is_ragged_input)
```

Comment:
Following your comment on line 172, this should also be changed to: `nest.flatten(is_ragged_input)`.